### PR TITLE
[SPARK-13574][SQL] Add benchmark to measure string dictionary decode.

### PR DESCRIPTION
**Pull request by nongli**
## What changes were proposed in this pull request?

Also updated the other benchmarks when the default to use vectorized decode was flipped.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetReadBenchmark.scala [hunk 1]
```diff
@@ -72,7 +72,7 @@ object ParquetReadBenchmark {
             .write.parquet(dir.getCanonicalPath)
         sqlContext.read.parquet(dir.getCanonicalPath).registerTempTable("tempTable")
 
-        sqlBenchmark.addCase("SQL Parquet Reader") { iter =>
+        sqlBenchmark.addCase("SQL Parquet Vectorized") { iter =>
           sqlContext.sql("select sum(id) from tempTable").collect()
```
Comment (davies) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetReadBenchmark.scala]: Could we also enable PARQUET_VECTORIZED_READER_ENABLED here?

Today I just ran into a issue when I have a conf/spark-default.conf that have PARQUET_VECTORIZED_READER_ENABLED in it.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetReadBenchmark.scala [hunk 2]
```diff
@@ -219,11 +216,43 @@ object ParquetReadBenchmark {
...
-        -------------------------------------------------------------------------------
-        SQL Parquet Reader                       1381 / 1679          7.6         131.7       1.0X
-        SQL Parquet MR                           2005 / 2177          5.2         191.2       0.7X
-        SQL Parquet Vectorized                    919 / 1044         11.4          87.6       1.5X
-        ParquetReader                            1035 / 1163         10.1          98.7       1.3X
+        -------------------------------------------------------------------------------------------
+        SQL Parquet Vectorized                   1025 / 1180         10.2          97.8       1.0X
+        SQL Parquet MR                           2157 / 2222          4.9         205.7       0.5X
+        SQL Parquet Non-vectorized               1450 / 1466          7.2         138.3       0.7X
+        ParquetReader                            1005 / 1022         10.4          95.9       1.0X
```
Comment (davies) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetReadBenchmark.scala]: Do you know why the first on is as fast as this one (only 20ms slower)?

Comment (nongli) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetReadBenchmark.scala]: The last one is the 1.6 parquet reader. What this is comparing, some what randomly, is the 1.6 reader without spark (last row) vs the vectorized reader with all of spark (first row).

Comment (davies) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetReadBenchmark.scala]: I see, could you append "Non-vectorized" after the name?

**Comment by davies** [comment]
LGTM

**Comment by davies** [comment]
Merging this into master, thanks!

**Pull request closed without merging**